### ISSUE TITLE
Adding CMake support

### ISSUE THREAD

**Issue by user404d** [shaih/HElib#181]
Are there any plans to add cmake support to this library? I know the library's main purpose is for research, but using cmake could make it easier for devs to jump in and contribute to the project. 

Are there issues with licensing or something?

**Comment by shaih** [comment]
Yes, it will be great to have CMake support, I just don't have the cycles to do it. Can anyone else do the job?

**Comment by fionser** [comment]
@user404d hope [this](https://github.com/shaih/HElib/pull/183) can help
